drafts/25-xxx-macro-expansion.txt

```diff
@@ -0,0 +1,517 @@
+      definition (section 4.3).
+
+ge04. During expansion, an identifier that is the name of a defined
+      function-like macro followed by a left parenthesis '(' as the next
+      token introduces an invocation of the function-like macro. The
+      arguments of the macro invocation are collected before the macro
+      is expanded.
+
+
+
+
+2 Function-like macro invocation
+================================
+
+
+2.1 Function-like macro identification
+--------------------------------------
+
+2.1.1 Argument gathering
+------------------------
+
+ag01. During expansion when a function-like macro name is encountered
+      followed by a '(' as the next token, the preprocessor shall scan
+      ahead, removing comments (section 2.1.4), to identify the full
+      list of macro arguments. The argument list begins after the
+      initial '(' and terminates at the matching closing parenthesis
+      ')'. The argument list must terminate by the end of the current
+      file.
+
+ag03. Commas within the argument list separate individual arguments.
+      However, a comma that appears within a balanced set of parentheses
+      (), square brackets '[]', curly braces '{}', or array constructor
+      delimiters '(/ /)' does not act as an argument separator. All such
+      brackets within an argument must be balanced and properly nested.
+
+TODO: Maybe add a straw poll here on the balance/nesting of bracketing?
+
+ag05. Commas and bracketing characters within comments are removed
+      (section 2.1.4) and thus do not affect argument separation.
+
+ag07. The collected arguments consist of all tokens between the
+      separating commas (or the opening/closing parentheses). Whitespace
+      surrounding an argument is not part of the argument itself. Line
+      continuations (sections 2.1.2 and 2.1.3), and comments (section
+      2.1.4) are removed as part of argument gathering and thus are not
+      part of any arguments themselves.
+
+ag09. For a macro that is not variadic (section 2.3), the number of
+      arguments in the invocation shall be equal to the number of
+      parameters in the macro definition.
+
+ag11. For a variadic macro (section 2.3), the number of arguments shall
+      not be less than the number of named parameters, and any arguments
+      supplied beyond the named parameters, along with the commas that
+      separate them, are gathered together as a single argument.
+
+ag13. An argument is considered empty if no preprocessing tokens appear
+      between its delimiters (e.g., between two commas, or between a
+      parenthesis and a comma). When an empty argument is substituted,
+      it is replaced by no tokens.
+
+ag15. Preprocessor directives are not recognized as such within the
+      arguments of a function-like macro invocation. Within the
+      arguments of a function-like macro invocation, the token INCLUDE
+      is not treated as introducing an INCLUDE line, even if it
+      resembles one.
```

Comment:
This seems worthy of two separate rules.

Also clarifying that we mean during argument gathering

```suggestion
      arguments of a function-like macro invocation. 
      
ag17. Within the
      arguments of a function-like macro invocation, the token INCLUDE
      is not treated as introducing an INCLUDE line during argument
      gathering, even if it resembles one.
```